mdbenchmark/plot.py

```diff
@@ -0,0 +1,188 @@
+    slope, intercept = calc_slope_intercept(
+        (df[df_sel].iloc[0], df['ns/day'].iloc[0]), (df[df_sel].iloc[1],
+                                                     df['ns/day'].iloc[1]))
+    # avoid a label and use values instead of pd.Series
+    ax.plot(
+        df[df_sel],
+        lin_func(df[df_sel].values, slope, intercept),
+        ls='--',
+        color=color,
+        alpha=.5)
+    return ax
+
+
+def plot_over_group(df, plot_cores, ax=None):
+    # plot all lines
+    df_sel = 'ncores' if plot_cores else 'nodes'
+
+    gb = df.groupby(['gpu', 'module', 'host'])
+    groupby = ['gpu', 'module', 'host']
+    for key, df in gb:
+        label = ' '.join(['{}={}'.format(n, v) for n, v in zip(groupby, key)])
+        plot_line(df=df, df_sel=df_sel, ax=ax, label=label)
+
+    # style axes
+    xlabel = 'cores' if plot_cores else 'nodes'
+    ax.set_xlabel('Number of {}'.format(xlabel))
+    ax.set_ylabel('Performance [ns/day]')
+    ax.legend()
+
+    return ax
+
+
+@cli.command()
+@click.option(
+    '--csv',
+    help='name of csv file',
+    multiple=True,
+    show_default=True)
+@click.option(
+    '--output-name',
+    '-o',
+    default=None,
+    help="name of output files",
+    show_default=True)
+@click.option(
+    '--output-type',
+    '-t',
+    help="file extension for plot outputs",
+    type=click.Choice(['png', 'pdf', 'svg', 'ps']),
+    show_default=True,
+    default='png')
+@click.option(
+    '--host-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="hostname",
+    show_default=True)
+@click.option(
+    '--module-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="module name or engine name (gromacs, namd)",
+    show_default=True)
+@click.option(
+    '--host-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="hostname",
+    show_default=True)
+@click.option(
+    '--gpu/--no-gpu',
+    help="plot data for GPU runs",
+    show_default=True,
+    default=False)
+@click.option(
+    '--cpu/--no-cpu',
+    help="plot data for CPU runs",
+    show_default=True,
+    default=True)
+@click.option(
+    '--plot-cores',
+    help="Plot performance per core instead of nodes",
+    show_default=True,
+    is_flag=True)
+def plot(csv, output_name, output_type, host_name, module_name, gpu, cpu, plot_cores):
+    """Generate plots from csv files"""
+
+    if len(csv) == 0:
+        console.error("No csv file specified. use --csv to specify a file.")
+
+    df = pd.concat([pd.read_csv(c, index_col=0) for c in csv]).dropna()
+
+    # gpu/cpu can be plotted together or separately
+    if gpu and cpu:
+        # this is incase the user decides to set both flags to true
+        console.info("plotting GPU and CPU data.")
+    elif gpu:
+        df = df[df.gpu]
+        console.info("plotting GPU data only.")
+    elif cpu:
+        df = df[~df.gpu]
+        console.info("plotting GPU data only.")
+    else:
+        console.info("plotting GPU and CPU data.")
```

Comment:
Does this case ever trigger? Also we should validate, if `cpu` and `gpu` are both false. In this case, we throw an error. Similar to what is implemented here: https://github.com/bio-phys/MDBenchmark/pull/69/files#diff-5ecad2076077c8bab2ad9de9fb25c45aR51